MSVC error in 'extras/alloc-pool/duk_alloc_pool.c'

Hello, just discovered Duktape and while playing around with the examples and
compiling and building `examples/hello/hello_ptrcomp.c` (which needs `extras/alloc-pool/duk_alloc_pool.c`),
the MSVC-2019 compiler throws this error:
```c
duk_alloc_pool.c(305): error C2036: 'void *': unknown size
gnumake: *** [Makefile.Windows:63: objects/duk_alloc_pool.obj] Error 2
```
I simply fixed it by:
```diff
--- a//duk_alloc_pool.c 2021-05-02 10:46:56
+++ b/duk_alloc_pool.c 2021-05-02 10:41:20
@@ -302,7 +302,7 @@
                } else {
                        waste_bytes += duk__alloc_pool_get_waste_estimate(curr, s->size);
                }
-               curr = curr + s->size;
+               curr = (void*) ((char*)curr + s->size);
        }
        used_count = (size_t) (s->count - free_count);
```

(I'm amazed this hasn't been fixed before and that Microsoft hasn't fixed it's product).

--- a/extras/alloc-pool/duk_alloc_pool.c
+++ b/extras/alloc-pool/duk_alloc_pool.c
@@ -302,7 +302,7 @@ void duk_alloc_pool_get_pool_stats(duk_pool_state *s, duk_pool_stats *res) {
 		} else {
 			waste_bytes += duk__alloc_pool_get_waste_estimate(curr, s->size);
 		}
-		curr = curr + s->size;
+		curr = (void *) ((duk_uint8_t *) curr + s->size);
 	}
 	used_count = (size_t) (s->count - free_count);
 

--- a/releases/releases.yaml
+++ b/releases/releases.yaml
@@ -1379,3 +1379,4 @@ duktape_releases:
     - "Improve DUK_USE_OS_STRING for macOS, iOS, watchOS, and tvOS (GH-2288)"
     - "Fix JSON.stringify() handling of Array 'replacer' duplicates (e.g. JSON.stringify({foo: 123}, [\"foo\", \"foo\"])); previously incorrectly serialized multiple times, now only once (GH-2379)"
     - "Use yaml.safe_load() instead of yaml.load() in Py2 tooling (GH-2384)"
+    - "Fix void pointer arithmetic in duk_alloc_pool.c, caused problems with MSVC (GH-2404, GH-2406)"

